src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VALUE = 10;
+    private static final double MORE_SIZE = 1.5;
+
+    private T[] array;
+
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_VALUE];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != size) {
+            checkIndex(index);
+        }
+        ensureCapacity();
+        size++;
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Object[] tempArray = new Object[size - index - 1];
+        final T removedElement = array[index];
+        System.arraycopy(array, index + 1, tempArray, 0, tempArray.length);
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        System.arraycopy(tempArray, 0, array, index, tempArray.length);
```

Comment:
one system arraycopy is enough for each method, hint: you can use the same array as both src and dest